### PR TITLE
Schedules page crashes to a black screen: labelless `assigned-sweep` catalog entry hits `null.map` in DefaultJobs

### LINKED ISSUE #827: Schedules page crashes to a black screen: labelless `assigned-sweep` catalog entry hits `null.map` in DefaultJobs
## Summary

Opening `/schedules` blanks the entire app (a black screen, nothing loads). The Default jobs tab renders every catalog entry, and the `assigned-sweep` default is a sweep that carries **no labels** (its selector is `assigned`, not a label), so the backend sends `labels: null`. The Default jobs render path calls `entry.labels.map(...)` for any sweep entry without a null guard, so `null.map` throws a `TypeError` during the first render. There is no app-level ErrorBoundary, so the throw unmounts the whole React tree and the page goes blank.

## Impact

- Every user on an instance whose catalog includes `assigned-sweep` (shipped in #799) cannot open the Schedules page at all. The Default jobs tab is the default tab, so the crash happens on load with no interaction.
- It does not matter whether the user has enabled `assigned-sweep`: the Default jobs tab renders all catalog entries regardless of enablement, so the entry's summary row alone crashes the page.

## Root cause

`web/src/components/DefaultJobs.tsx:308` (options cell, runs on initial render):

```tsx
{entry.target === "sweep" && entry.labels.map((l) => <Chip key={l}>label {l}</Chip>)}
```

For `assigned-sweep`, `entry.target === "sweep"` is true and `entry.labels` is `null`, so this evaluates `null.map(...)` and throws `TypeError: Cannot read properties of null (reading 'map')`.

The web `CatalogEntry.labels` type is `string[]` (`web/src/lib/api.ts:1300`), which is wrong: the backend now emits `null` for a sweep whose selector is not label-based. The non-null type is why no optional chaining guards the access.

## Evidence

- Live catalog: `assigned-sweep` has `target: "sweep"` and `labels: null`. Every other sweep default (`bug-triage`, `planned-sweep`) carries a real `labels` array and renders fine.
- The entry landed in #799; the backend test `api/internal/schedtmpl/schedtmpl_test.go:143` pins that `assigned-sweep` must carry no labels.
- No ErrorBoundary anywhere in `web/src`: `web/src/App.tsx` wraps nothing, and the tree has no production ErrorBoundary at all (`RunView.tsx` even documents this in a comment). So a render throw on this page unmounts the whole React tree.

## Why tests and mock mode did not catch it

The mock schedule catalog (the `scheduleCatalog` array in `web/src/mocks/mockApi.ts`) contains no labelless sweep at all: its two sweeps carry non-empty label arrays (`["bug"]`, `["Planned"]`) and there is no `assigned-sweep` entry. So no fixture ever produces `labels: null`, the route smoke test renders the Default jobs tab green, and the crashing branch is never exercised. The mock API layer itself spreads `[...entry.labels]` when building catalog rows, so it too would throw on a null-labels entry, which is why the current fixtures cannot model this bug.

## Related latent bug (same null, not the initial crash)

`web/src/components/DefaultJobs.tsx:92`, in the enable handler:

```tsx
if (entry.target === "sweep" && entry.labels.length > 0) {
```

This reads `null.length` and would also throw, but only when a user clicks Enable on `assigned-sweep`.

`web/src/pages/Schedules.tsx:952` already guards the same shape correctly (`s.labels && s.labels.length > 0`), which is the pattern the two `DefaultJobs` sites are missing.

## Proposed fix (smallest correct change)

1. Fix the type: `CatalogEntry.labels` should be `string[] | null` in `web/src/lib/api.ts`, matching the backend. Note this makes `labels` possibly-null everywhere it is read, including two unguarded spreads in the mock layer (`web/src/mocks/mockApi.ts`, the `[...entry.labels]` / `[...e.labels]` sites), which must be null-guarded or `typecheck:web` (part of `task gate:web`) will go red.
2. Guard both `DefaultJobs.tsx` sites against a null `labels`:
   - line 308: render label chips only when `entry.labels` is a non-empty array (for example `entry.target === "sweep" && (entry.labels?.length ?? 0) > 0 && entry.labels.map(...)`, or normalize with `(entry.labels ?? [])`).
   - line 92: guard the length read the same way (`entry.labels && entry.labels.length > 0`). This also covers the `labels: entry.labels` read at line 97, which sits inside the same guard.
3. Add a regression test: exercise the Default jobs tab with a labelless sweep catalog entry (`labels: null`) and assert it renders without throwing. Construct the `CatalogEntry` directly and mount `DefaultJobs` (or null-guard the mock catalog layer from step 1 first), because routing a `labels: null` entry through the mock `listScheduleCatalog` currently throws in the mock's own `[...e.labels]` spread before `DefaultJobs` renders. The test should fail before the fix and pass after.

Keep the change scoped to null-labels handling (the two `DefaultJobs` sites, the type, the two mock-layer spreads, and the test). An app-level ErrorBoundary would have contained the blast radius, but it is separate hardening and out of scope for this fix.

### PR DESCRIPTION
**Pull request by vtmocanu-uzi**
Implements issue #827.

Closes #827

> ⚠️ This run used agent definitions from the repository's own `.claude/agents/` (architect, auditor, coder, documenter, fact-checker, release, researcher, reviewer, spec-keeper, tester, tui-ux, ux-designer, web-ux). The internal review was performed by those repo-authored agents, not by uzi's built-in reviewer — review this change accordingly.

---
Opened automatically by the uzi agent from branch `agent/issue-827`. Please review and merge manually — the agent never merges.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed assigned sweeps without labels so they render correctly without crashing.
  * Prevented empty label chips from appearing when no labels are configured.
  * Preserved support for sweeps that select issues by assignee instead of label.
  * Catalog entries now accurately retain whether label selection is configured.

* **Tests**
  * Added regression coverage for assigned sweeps with no labels.
  * Updated catalog coverage for the new assigned-sweep scenario.

### CODE REVIEW

File: web/src/components/DefaultJobs.test.tsx [hunk 1]
```diff
@@ -213,6 +213,20 @@ describe("DefaultJobs — catalog row", () => {
...
   });
 
+  it("renders a labelless assigned sweep without crashing", () => {
+    // Regression for the /schedules black-screen: the shipped assigned-sweep default carries
+    // labels: null, and the options cell read entry.labels.map() unguarded, so the tab threw
+    // on first render. Rendering the row must not throw, and with no labels no "label …" chip
+    // is emitted.
+    renderTab({
+      catalog: catalog([entry({ slug: "assigned-sweep", name: "Assigned sweep", labels: null })]),
+    });
```
[the remark this answers is not in the document]
Comment (vtmocanu-uzi) [reply web/src/components/DefaultJobs.test.tsx:223]: Done in cfc11e8f. Added a test ("Enable on a labelless assigned sweep runs the null-guarded enable handler without throwing") in DefaultJobs.test.tsx that renders the labelless `assigned-sweep` entry, selects a repo, clicks the row Enable button, and asserts `onEnable` is called with the labelless entry (`labels: null`) and the selected repo id, and that the row remains rendered afterward. This now exercises the `entry.labels && entry.labels.length > 0` guard in `onEnableRow` — fold-verified: removing the `entry.labels &&` guard reddens the run (`TypeError: Cannot read properties of null (reading 'length')`), which the prior render-only test did not catch.

File: web/src/mocks/scheduleCatalog.test.ts [hunk 2]
```diff
@@ -9,10 +9,10 @@ import { describe, expect, it } from "vitest";
 import { mockApi } from "./mockApi";
 
 describe("mock default-jobs catalog (PRD #589)", () => {
-  it("lists the 7 catalog entries and seeds a visible Layout-A demo state", async () => {
+  it("lists the 8 catalog entries and seeds a visible Layout-A demo state", async () => {
     const { entries, enablements } = await mockApi.listScheduleCatalog();
     expect(entries.map((e) => e.slug).sort()).toEqual(
-      ["bug-hunt", "bug-triage", "docs-hygiene", "feature-bingo", "planned-sweep", "self-improve", "test-improvement"].sort(),
+      ["assigned-sweep", "bug-hunt", "bug-triage", "docs-hygiene", "feature-bingo", "planned-sweep", "self-improve", "test-improvement"].sort(),
```
[the remark this answers is not in the document]
Comment (vtmocanu-uzi) [reply web/src/mocks/scheduleCatalog.test.ts:15]: Done in cfc11e8f. Added `expect(entries.find((e) => e.slug === "assigned-sweep")?.labels).toBeNull();` to the catalog-listing test, alongside the existing slug assertion — so a regression from `labels: null` to `labels: []` in the mock catalog now fails the test.
